textile/fabric_printing/report/coating_production_register/coating_production_register.py

```diff
@@ -0,0 +1,316 @@
+# Copyright (c) 2023, ParaLogic and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _, scrub, unscrub
+from frappe.utils import cint, cstr, flt, getdate, add_days
+from frappe.desk.query_report import group_report_data
+from erpnext.setup.doctype.uom_conversion_factor.uom_conversion_factor import get_uom_conv_factor
+
+
+def execute(filters=None):
+	return CoatingProductionRegister(filters).run()
+
+
+class CoatingProductionRegister:
+	def __init__(self, filters=None):
+		self.filters = frappe._dict(filters or {})
+		self.filters.from_date = getdate(filters.from_date)
+		self.filters.to_date = getdate(filters.to_date)
+
+		if self.filters.from_date > self.filters.to_date:
+			frappe.throw(_("Date Range is incorrect"))
+
+		self.show_item_name = frappe.defaults.get_global_default('item_naming_by') != "Item Name"
+		self.show_customer_name = frappe.defaults.get_global_default('cust_master_name') == "Naming Series"
+
+	def run(self):
+		self.get_data()
+		self.prepare_data()
+		self.get_chart_data()
+		self.data = self.get_grouped_data()
+		self.get_columns()
+
+		return self.columns, self.data, None, self.chart_data
+
+	def get_data(self):
+		conditions = self.get_conditions()
+
+		self.data = frappe.db.sql("""
+			SELECT se.name as stock_entry, se.posting_date, se.posting_time,
+				timestamp(se.posting_date, se.posting_time) as posting_dt,
+				se.coating_order, se.fg_completed_qty as qty,
+				co.uom as uom,
```

Comment:
Wrong UOM. Stock Entry uses qty as per stock uom instead of selected uom.. use `co.stock_uom as uom`

we are expecting to get Meters .. but just in case if it's not in Meters, the report will convert it to Meters